src/utils/license.ts

```diff
@@ -1,47 +1,45 @@
-import type {
-  License,
-  PublicDomainMark,
-  LicenseVersion,
-  CcLicense,
-} from '~/constants/license'
+import type { License, LicenseVersion } from '~/constants/license'
 import {
   CC_LICENSES,
   DEPRECATED_CC_LICENSES,
   PUBLIC_DOMAIN_MARKS,
 } from '~/constants/license'
 
-// type LicenseName =
-//   | `${Uppercase<PublicDomainMark>} 1.0`
-//   | `CC ${Capitalize<DeprecatedCcLicense>} 1.0`
-//   | `CC ${Uppercase<CcLicense>} ${LicenseVersion}`
-
-type LicenseName =
-  | `${Uppercase<PublicDomainMark>} ${LicenseVersion}` // technically only v1
-  | `CC Sampling+ ${LicenseVersion}` // technically only v1
-  | `CC NC-Sampling+ ${LicenseVersion}` // technically only v1
-  | `CC ${Uppercase<CcLicense>} ${LicenseVersion}`
+import type { IVueI18n } from 'vue-i18n'
 
 /**
  * Get the full name of the license in a displayable format from the license
  * slug and version.
  *
  * @param license - the slug of the license
  * @param licenseVersion - the version number of the license
+ * @param i18n - the i18n instance to access translations
  * @returns the full name of the license
  */
 export const getFullLicenseName = (
   license: License,
-  licenseVersion: LicenseVersion = '' // unknown version
-): LicenseName => {
-  if ((PUBLIC_DOMAIN_MARKS as ReadonlyArray<License>).includes(license)) {
-    const licenseUpper = license.toUpperCase() as Uppercase<PublicDomainMark>
-    return `${licenseUpper} ${licenseVersion}`
+  licenseVersion: LicenseVersion = '', // unknown version
+  i18n: IVueI18n | null = null
```

Comment:
This can be `null` if you're invoking the function from outside a Vue component (thus don't have access to the `i18n` instance) and are okay with seeing PDM as the license name.

Reply:
Ideally I updated all occurrences of this function to provide this argument, but the function should still be usable if `i18n` was unavailable.